src/main/java/com/openshift/jenkins/plugins/pipeline/OcAction.java

```diff
@@ -156,100 +159,59 @@ protected OcActionResult run() throws IOException, InterruptedException, Executi
             if (filePath != null && !filePath.exists()) {
                 filePath.mkdirs();
             }
-            
+
             if (step.streamStdOutToConsolePrefix != null
                     && step.streamStdOutToConsolePrefix
-                        .startsWith("start-build")) {
+                    .startsWith("start-build")) {
                 listener.getLogger()
                         .println(
                                 "NOTE: the selector returned when -F/--follow is supplied to startBuild() will be inoperative for the various selector operations.");
                 listener.getLogger()
                         .println(
                                 "Consider removing those options from startBuild and using the logs() command to follow the build output.");
             }
-
-            ByteArrayOutputStream stdErr = new ByteArrayOutputStream();
-            BufferedOutputStream errBuf = new BufferedOutputStream(stdErr);
-            ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-            BufferedOutputStream outBuf = new BufferedOutputStream(stdOut);
-
             String commandString = step.cmdBuilder.asString(false);
             String[] command = QuotedStringTokenizer.tokenize(commandString);
             command = ClientCommandBuilder.fixPathInCommandArray(command, envVars, listener, filePath, launcher, step.verbose);
-
-            Proc proc = null;
+            final StringBuffer stdout = new StringBuffer();
+            final StringBuffer stderr = new StringBuffer();
+            ClientCommandRunner runner = new ClientCommandRunner(command, filePath, envVars,
+                    line -> { // got a line from stdout
+                        stdout.append(line).append('\n');
+                        printToConsole(line);
+                        return false; // don't interrupt `oc`
+                    },
+                    line -> { // got a line from stderr
+                        stderr.append(line).append('\n');
+                        printToConsole(line);
+                        return false; // don't interrupt `oc`
+                    });
+
+            int exitStatus = -1;
             try {
-                Launcher.ProcStarter ps = launcher.launch().cmds(Arrays.asList(command)).envs(envVars).pwd(filePath).quiet(true).stdout(outBuf).stderr(errBuf);
-                proc = ps.start();
-                long reCheckSleep = 250;
-                int outputSize = 0;
-                while (proc.isAlive()) {
```

Comment:
I do *REALLY* like getting rid of this loop :-) .... much cleaner with your new way

Reply:
Haha, it would be nice if we can really get rid of the loop.